compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -2419,6 +2425,20 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             );
         }
 
+        derives_grouped
```

Comment:
Could you move the code responsible for grouping of the derives from `note_pred_src` to `suggest_derive`?